app/kubernetes/pod/payloads/create.js

```diff
@@ -0,0 +1,52 @@
+import { KubernetesCommonMetadataPayload } from 'Kubernetes/models/common/payloads';
+
+/**
+ * CreatePayload Model
+ */
+const _CreatePayload = Object.freeze({
+  metadata: new KubernetesCommonMetadataPayload(),
+  spec: {
+    replicas: 0,
+    selector: {
+      matchLabels: {
+        app: '',
+      },
+    },
+    strategy: {
+      type: 'RollingUpdate',
+      rollingUpdate: {
+        maxSurge: 0,
+        maxUnavailable: '100%',
+      },
+    },
+    template: {
+      metadata: {
+        labels: {
+          app: '',
+        },
+      },
+      spec: {
+        affinity: {},
+        containers: [
+          {
+            name: '',
+            image: '',
+            env: [],
+            resources: {
+              limits: {},
+              requests: {},
+            },
+            volumeMounts: [],
+          },
+        ],
+        volumes: [],
+      },
+    },
+  },
+});
+
+export class CreatePayload {
+  constructor() {
+    Object.assign(this, JSON.parse(JSON.stringify(_CreatePayload)));
```

Comment:
Note that this way of deep cloning will cause data lose. I'd recommend [this](https://frontbackend.com/javascript/what-is-the-the-fastest-way-to-deep-clone-an-object-in-javascript) article for comparing different ways of doing deep clone.

Reply:
I agree, I don't like this pattern. This is how it was done in the kube PR, and this PR is mostly adjusting stuff of deployments to a pod. I'll replace with a factory function

Reply:
I think in the missed part about deep cloning that the article missed is just not to do it :)